packages/x-data-grid/src/components/GridRow.tsx

```diff
@@ -515,7 +518,11 @@ const GridRow = React.forwardRef<HTMLDivElement, GridRowProps>(function GridRow(
       {...other}
     >
       {leftCells}
-      <div className={gridClasses.cellOffsetLeft} role="presentation" />
+      <div
```

Comment:
Could you mention the steps you are doing to test it, in a few of my performance snapshots I could see a noticeable reduction in the recalculate styles portion, but it regresses a bit the layout part of the rendering.
I took into account those style revalidations which came after the scroll event.

Here are the average results from a few tests on the M1 Pro and Chrome Version 121.0.6167.184

**Before: (on https://next.mui.com/x/react-data-grid/#pro-plan)**
Recalculate style: 4.95ms
Layout: 0.91ms
Pre-paint: 0.91
Paint: 1.14ms
![image](https://github.com/mui/mui-x/assets/12609561/e8b9a9a6-2286-4113-ba85-a97671c8de36)


**After: (on https://deploy-preview-12019--material-ui-x.netlify.app/x/react-data-grid/#pro-plan)**
Recalculate style: 2.91ms
Layout: 7.75ms
Pre-paint: 0.90
Paint: 0.80ms
![image](https://github.com/mui/mui-x/assets/12609561/b62ab438-dfe8-49c3-b8bc-93aaf344c29d)

Am I testing it incorrectly?

Bdw, on 4x CPU slowdown, it appears to improve the layout phase as well (almost 40% improvement), so overall it seems improved, especially on low-end devices. I am all in for pushing for the improvement!

But if you can guess the possible reason for the increase in the layout calculation time without throttling, that'd add to my knowledge.

Reply:
I run `scroller.scrollTo({ top: 100_000, behavior: 'smooth' })` and inspect one of the cycles in the middle of the graph. I'm not sure exactly, but it could be fixed in #12023. Using an `offsetTop` on the row container in sticky was a mistake, it triggers style invalidation, and it might be triggering layout invalidation.